beacon-chain/core/blocks/block_operations.go

```diff
@@ -440,6 +440,54 @@ func ProcessBlockAttestations(
 	return beaconState, nil
 }
 
+func verifyAttestationSig(beaconState *pb.BeaconState, att *pb.Attestation) error {
+	attesterIndices, err := helpers.AttestationParticipants(beaconState, att.Data, att.AggregationBitfield)
+	if err != nil {
+		return fmt.Errorf("could not get attestation participants: %v", err)
+	}
+
+	attestorPubKeys := make([]*bls.PublicKey, len(attesterIndices))
+	for idx, participantIndex := range attesterIndices {
+		pubkey, err := bls.PublicKeyFromBytes(beaconState.ValidatorRegistry[participantIndex].Pubkey)
+		if err != nil {
+			return fmt.Errorf("could not deserialize attestor public key: %v", err)
+		}
+
+		attestorPubKeys[idx] = pubkey
+	}
+
+	aggregatePubkeys := []*bls.PublicKey{
+		bls.AggregatePublicKeys(attestorPubKeys),
+	}
+
+	attestationDataHash, err := hashutil.HashProto(&pb.AttestationDataAndCustodyBit{
+		Data:       att.Data,
+		CustodyBit: true,
+	})
+	if err != nil {
+		return fmt.Errorf("could not hash attestation data: %v", err)
+	}
+	messageHashes := [][]byte{
+		attestationDataHash[:],
+	}
+
+	sig, err := bls.SignatureFromBytes(att.AggregateSignature)
+	currentEpoch := helpers.CurrentEpoch(beaconState)
+
+	domain := forkutils.DomainVersion(beaconState.Fork, currentEpoch, params.BeaconConfig().DomainAttestation)
+
+	log.WithFields(logrus.Fields{
+		"aggregatePubkeys": fmt.Sprintf("%#x", aggregatePubkeys),
```

Comment:
this will be an extremely big log if we have many attesters , same with the one with message hashes. Just logging the aggregate signature is fine